dandiapi/api/views/dandiset.py

```diff
@@ -154,6 +156,65 @@ def get_object(self):
                 raise PermissionDenied()
         return dandiset
 
+    @swagger_auto_schema(
+        query_serializer=DandisetQueryParameterSerializer(),
+        responses={200: DandisetListSerializer(many=True)},
+    )
+    def list(self, request, *args, **kwargs):
+        qs = self.get_queryset()
+        dandisets = self.paginate_queryset(self.filter_queryset(qs))
+
+        # Query versions, to supply size/count of assets.
+        base_query = (
+            Version.objects.select_related('dandiset')
+            .annotate(
+                num_assets=Count('assets'),
+                total_size=(
+                    Coalesce(Sum('assets__blob__size'), 0)
+                    + Coalesce(Sum('assets__embargoed_blob__size'), 0)
+                    + Coalesce(Sum('assets__zarr__size'), 0)
+                ),
+            )
+            .filter(dandiset__in=dandisets)
+            .order_by('-version', '-modified')
+        )
+
+        latest_dandiset_version = (
+            Version.objects.exclude(version='draft')
+            .order_by('-version')
+            .filter(dandiset_id=OuterRef('dandiset_id'))
+            .values('version')[:1]
+        )
+        published = (
+            base_query.exclude(version='draft')
+            .alias(latest=Subquery(latest_dandiset_version))
+            .filter(version=F('latest'))
+        )
+        drafts = base_query.filter(version='draft')
+
+        # Union with drafts
+        versions = published.union(drafts).order_by('dandiset_id', '-version')
```

Comment:
@mvandenburgh I think I figured out how to slim down the list of versions to just one draft version and one published version (if it exists) per dandiset. I wish I could make this useful to more places in the codebase than just here. If you can think of anywhere else this might apply please lmk.